libfwupdplugin/fu-device.c

```diff
@@ -3840,15 +3842,29 @@ fu_device_get_backend_parent_with_subsystem(FuDevice *self, const gchar *subsyst
 		const gchar *gtype_str;
 		const gchar *id;
 
-		/* we have to propagate this to preserve compat with older emulation files */
-		event = fu_device_load_event(FU_DEVICE(self), event_id, NULL);
-		if (event == NULL) {
-			g_debug("falling back to simulated device for old emulation");
-			parent =
-			    fu_backend_get_device_parent(priv->backend, self, subsystem, error);
-			if (parent != self)
-				fu_device_set_target(parent, self);
-			return g_steal_pointer(&parent);
+		/* we might have to propagate this to preserve compat with older emulation files */
+		if (priv->fwupd_version != NULL &&
+		    fu_version_compare(priv->fwupd_version,
+				       "2.0.8",
+				       FWUPD_VERSION_FORMAT_TRIPLET) >= 0) {
+			event = fu_device_load_event(FU_DEVICE(self), event_id, error);
+			if (event == NULL)
+				return NULL;
+			if (!fu_device_event_check_error(event, error))
+				return NULL;
+		} else {
+			event = fu_device_load_event(FU_DEVICE(self), event_id, NULL);
+			if (event == NULL) {
+				g_debug("falling back for emulation recorded by fwupd %s",
+					priv->fwupd_version);
+				parent = fu_backend_get_device_parent(priv->backend,
+								      self,
+								      subsystem,
+								      error);
+				if (parent != self)
+					fu_device_set_target(parent, self);
+				return g_steal_pointer(&parent);
+			}
```

Comment:
presumably we want some tests for explicitly checking both, right?

Reply:
I think we get that for free with the enumerations tbh.